docs/source/custom_task.rst

```diff
@@ -56,37 +86,152 @@ override ``step`` (which is called for training, validation, and
 testing) or override ``training_step``, ``validation_step``, and
 ``test_step`` individually. These methods behave identically to PyTorch
 Lightning’s
-`methods <https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#methods>`__.
+`methods <https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#methods>`__.
 
-The Data
---------
+Here is the pseudo code behind :class:`~flash.core.model.Task` step.
 
-For a task you will likely need a specific way of loading data. For this
-example, lets say we want a ``flash.DataModule`` to be used explicitly
-for the prediction of diabetes disease progression. We can create this
-``DataModule`` below, wrapping the scikit-learn `Diabetes
-dataset <https://scikit-learn.org/stable/datasets/toy_dataset.html#diabetes-dataset>`__.
+Example::
+
+    def step(self, batch: Any, batch_idx: int) -> Any:
+        """
+        The training/validation/test step. Override for custom behavior.
+        """
+        x, y = batch
+        y_hat = self(x)
+        # compute the logs, loss and metrics as an output dictionary
+        return output
+
+
+3.a The DataModule API
+----------------------
+
+First, let's first design the user-facing API.
```

Comment:
Start with high-level. Something like:

Now that we have defined the task, we need to load our data. We will define a custom Flash :class:...DataModule to load the data from numpy arrays, and then define the preprocessing steps in a custom Flash :class:....preprocess.

a. The DataModule
Flash uses DataModules to load the datasets. Since our dataset is stored as a.... we will need to create a custom datamodule that will take that as input. We are going to create a NumpyDataModule....


b...